lib/controllers/facebook-login.js

```diff
@@ -60,21 +60,27 @@ module.exports = function (req, res) {
 
       res.locals.user = expandedAccount;
       req.user = expandedAccount;
-      helpers.createIdSiteSession(req.user, req, res);
 
-      var nextUrl = url.parse(req.query.next || '').path || (resp.created ? config.web.register.nextUri : config.web.login.nextUri);
+      helpers.createStormpathSession(req.user, req, res, function (err) {
+        if (err) {
+          logger.info('During a Facebook OAuth login attempt, we were unable to create a Stormpath session.');
+          return res.status(err.status || 400).json(err);
+        }
 
-      if (resp.created && registrationHandler) {
-        registrationHandler(req.user, req, res, function () {
-          res.redirect(302, nextUrl);
-        });
-      } else if (loginHandler) {
-        loginHandler(req.user, req, res, function () {
+        var nextUrl = url.parse(req.query.next || '').path || (resp.created ? config.web.register.nextUri : config.web.login.nextUri);
```

Comment:
A suggestion:

This line is very long and hard to follow. Considering breaking it up to something more readable, like this:

```javascript
var nextUrlConfig = resp.created ? config.web.register.nextUri : config.web.login.nextUri;
var nextUrlQuery = url.parse(req.query.next || '').path;
var nextUrl = nextUrlQuery || nextUrlConfig;
```